spec/controllers/idv/phone_errors_controller_spec.rb

```diff
@@ -1,7 +1,6 @@
 require 'rails_helper'
 
-RSpec.describe Idv::PhoneErrorsController,
-               allowed_extra_analytics: [:sample_bucket1, :sample_bucket2] do
+RSpec.describe Idv::PhoneErrorsController, allowed_extra_analytics: [:idv_phone_error_visited] do
```

Comment:
Just to confirm: before, the `allowed_extra_analytics` was individual keys of methods, but this is now switching to method names. Do we think that it will be all-or-nothing for entire methods?

Reply:
As far as I can tell, it's always been about allowlisting methods, so I'm not really sure why this worked previously. I'll dig into it.

Reply:
no, `sample_bucket1` and `sample_bucket2` were individual keys used only in specs for some of the IDV analytics methods

Reply:
Yes, but `UndocumentedParamsChecker` is supposed to be checking the array as an array of method names.

https://github.com/18F/identity-idp/blob/44c9479d1f62cd4ddca4dc00d57ab6fe25c1e02f/spec/support/fake_analytics.rb#L87

Reply:
Oh, I see now that it _also_ checks the parameters as well, I missed that before.

Although... do we need it?

https://github.com/18F/identity-idp/blob/44c9479d1f62cd4ddca4dc00d57ab6fe25c1e02f/spec/support/fake_analytics.rb#L97

Reply:
In my intended use case, I wanted to know what keys were missing from events, because that was labor-intensive to trace the keys back to every single callsite, and because for a few AB-test example buckets, it was clear to me we didn't need to document the sample values.

We could keep both if you want, by updating to a hash structure for the spec metadata:
```
allowed_extra_analytics: {
  method_name: [:param1, :param2],
  other_method_name: :*, # single-method allow-all
  :* => :*, # all methods
}
```

Reply:
Ok, 2f5ee92 reverts back to the original intention of `allowed_extra_analytics`.

After discussing in Slack, it looks like the root of this stems back to a misunderstanding of how `allowed_extra_analytics` was intended to be used when enhancing it in my own pull request #10252.